### PR TITLE
Fixup package

### PR DESCRIPTION
**Pull request by Jake-Shadle**
This fixes up the `package.exclude` field to be simpler and actually exclude everything that is unused for downstream users, notably the entirety of the `scripts` folder. The `target/*` and `Cargo.lock` values were superfluous since they are already present in the .gitignore which cargo excludes by default.

While I was doing this I noticed that github CI was now used, but the .travis.yml file was still in the root, and the README still used the travis badge, so fixed that.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ This crate provides Grapheme Cluster, Word and Sentence boundaries
 according to Unicode Standard Annex #29 rules.
 """
 
-exclude = [ "target/*", "Cargo.lock", "scripts/tmp", "benches/texts/*", "*.txt", ]
+exclude = ["/.github", "/scripts", "/benches/texts"]
```
Comment (Manishearth) [inline Cargo.toml:20]: I'd rather not exclude `scripts`, it's not used by the Rust package but people vendoring the crate sometimes care about having files like this. Can this be reverted to excluding `scripts/tmp`?

Comment (Jake-Shadle) [reply Cargo.toml:20]: Ahh ok, that's unfortunate, the main motivation for the change was to remove executable scripts, but I'll revert that part.

Comment (kwantam) [reply Cargo.toml:20]: What's the reason for removing executable scripts? In general I'm 100% sympathetic to the idea that one shouldn't ship unnecessary executables (and I'm aware of the recent kerfuffle in the broader community), but in this case (1) they're scripts, so they're easy to inspect and compare against the git history, and (2) they are not, to my knowledge, executed automatically under any circumstances.

Comment (Manishearth) [reply Cargo.toml]: @Jake-Shadle if they have an executable bit set I do not mind removing that, but it's not a huge deal. They are scripts, not binaries.

File: Cargo.toml [hunk 2]
```diff
@@ -17,7 +17,7 @@ This crate provides Grapheme Cluster, Word and Sentence boundaries
 according to Unicode Standard Annex #29 rules.
 """
 
-exclude = ["/.github", "/scripts", "/benches/texts"]
+exclude = ["/.github", "/benches/texts"]
```
Comment (Manishearth) [inline Cargo.toml:20]: Actually, while you're at it, could you exclude: `*.txt`, `scripts/*.rs`. Those things we *do* want to exclude (the first is downloaded unicode data, the second is temporary rust files)

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
Thanks!

**Pull request merged**